packages/thirdweb/src/react/core/hooks/transaction/transaction-button-utils.ts

```diff
@@ -113,14 +117,66 @@ export const useTransactionButtonMutation = (
     onError,
     onClick,
   } = props;
+
+  // yep, a state.
+  const [resolvedTransaction, setResolvedTransaction] =
+    useState<PreparedTransaction>();
+
+  // yep, a useEffect
+  useEffect(() => {
+    // if we are already resolving this transaction, don't do it again
+    if (TX_RESOLVE_WEAK_SET.has(transaction)) {
+      return;
+    }
+    let active = true;
+    TX_RESOLVE_WEAK_SET.add(transaction);
+    async function run() {
+      try {
+        const resolvedTx = await transaction();
+        // only set it if this was the last version of it
+        if (active) {
+          setResolvedTransaction(resolvedTx);
+        }
+      } catch {
+        // remove from set so we can retry later
+        TX_RESOLVE_WEAK_SET.delete(transaction);
+      }
+    }
+    // actually run the async thing
+    run();
+    // cleanup the active state if tx changes
+    return () => {
+      active = false;
+    };
+  }, [transaction]);
+
+  // pass the resolved transaction to the prepare transaction hook
+  const preparedTransactionMutation = usePrepareTransaction(
+    resolvedTransaction,
+    {
+      queryOptions: {
+        // we only want to run the query if we have a resolved transaction
+        enabled: !!resolvedTransaction,
+      },
+    },
+  );
+
   return useMutation({
     mutationFn: async () => {
       if (onClick) {
         onClick();
       }
+
+      // if we have not resolved the transaction yet, don't do anything
+      if (!preparedTransactionMutation.data) {
+        // not ready yet
+        return;
```

Comment:
we probably want to disable the button while the transaction is preparing?

trying to think of the deploy scenario:

1. i pass a prepareDeployTx() to TxButton
2. it prepares it for me on mount (uploads some metadata, takes a while)
3. i click deploy - nothing happens (cause its still uploading) and i get no feedback

few UX options here but yeah we need some sort of feedback